syntax = "proto3";

enum MovieCategory {
  ACTION = 0;
  COMEDY = 1;
  HORROR = 2;
  SCIENCE_FICTION = 3;
}

message RecommendationRequest {
  int32 user_id = 1;
  MovieCategory category = 2;
  int32 max_results = 3;
}

message MovieRecommendation {
  int32 id = 1;
  string title = 2;
}

message RecommendationResponse {
  repeated MovieRecommendation recommendations = 1;
}

service Recommendations {
  rpc Recommend (RecommendationRequest) returns (RecommendationResponse);
}